ethcore/src/snapshot/mod.rs

```diff
@@ -34,74 +31,117 @@ use util::{Bytes, Hashable, HashDB, TrieDB};
 use util::hash::{FixedHash, H256};
 use util::numbers::U256;
 use util::rlp::{DecoderError, Rlp, RlpStream, Stream, SHA3_NULL_RLP, UntrustedRlp, View};
+use util::snappy;
+
+use self::block::AbridgedBlock;
+
+mod block;
+
+// Try to have chunks be around 16MB (before compression)
+const PREFERRED_CHUNK_SIZE: usize = 16 * 1024 * 1024;
+
+// use initially 20MB for the reusable snappy buffers.
+// should always be larger than PREFERRED_CHUNK_SIZE for fault tolerance.
+const SNAPPY_BUFFER_SIZE: usize = 20 * 1024 * 1024;
```

Comment:
considering the buffer gets grown automatically if it is too small, it might just make sense to initially
allocate just `max_compressed_len(16MB)`

once we encode data to RLP, the overhead will often push it slightly over 16MB, so it may need growing once or twice, but this is unlikely.